### PR TITLE
Tiny cleanup nuts test

### PR DESCRIPTION
**Pull request by fehiepsi**
~This PR fixes various bugs reported in #1532 and #1511.~

This PR adds a clearer message for the reason why JIT is slow.

### CODE REVIEW

File: tests/infer/mcmc/test_nuts.py [hunk 1]
```diff
@@ -53,20 +53,20 @@
     ),
     T(
         GaussianChain(dim=5, chain_len=2, num_obs=10000),
-        num_samples=800,
-        warmup_steps=200,
+        num_samples=400,
+        warmup_steps=20,
```
Comment (fehiepsi) [inline tests/infer/mcmc/test_nuts.py]: this test is slow, so I reduce warmup_steps to the minimum of adaptation steps.

File: tests/infer/mcmc/test_hmc.py [hunk 2]
```diff
@@ -123,7 +123,6 @@ def rmse(t1, t2):
     'fixture, num_samples, warmup_steps, hmc_params, expected_means, expected_precs, mean_tol, std_tol',
     TEST_CASES,
     ids=TEST_IDS)
-@pytest.mark.skip(reason='Slow test (https://github.com/pytorch/pytorch/issues/12190)')
```
Comment (neerajprad) [inline tests/infer/mcmc/test_hmc.py]: I think we should skip this on CI for sure. All of these together add several minutes to testing time and provide little in the way of additional information (the code paths they test are covered by faster tests). They are very useful for local profiling though, so if they all take more than a minute I would keep the skip flag.

File: tests/infer/mcmc/test_nuts.py [hunk 3]
```diff
@@ -96,7 +96,6 @@ def jit_idfn(param):
     'fixture, num_samples, warmup_steps, expected_means, expected_precs, mean_tol, std_tol',
     TEST_CASES,
     ids=TEST_IDS)
-@pytest.mark.skip(reason='Slow test (https://github.com/pytorch/pytorch/issues/12190)')
```
Comment (neerajprad) [inline tests/infer/mcmc/test_nuts.py]: Same here - I think they can't be made much faster, and they will still add 5 mins to travis, so let us only keep these for local runs.

File: tests/infer/mcmc/test_hmc.py (hunk 2 above)
Comment (fehiepsi) [reply tests/infer/mcmc/test_hmc.py]: You are right! I should look at CI log before making this PR. So we'll keep this for profiling. I'll change the title for this PR though (clearer skip messages)

**Comment by neerajprad** [comment]
You'll need to merge the latest dev.

**Pull request merged**